templates/aws-stack.yml

```diff
@@ -1019,10 +1022,14 @@ Resources:
                         - kms:GetPublicKey
                       - - kms:Verify
                         - kms:GetPublicKey
-                  Resource: !If
-                              - CreatePipelineSigningKMSKey
-                              - !Sub arn:aws:kms:${AWS::Region}:${AWS::AccountId}:key/${PipelineSigningKMSKey}
-                              - !Sub arn:aws:kms:${AWS::Region}:${AWS::AccountId}:key/${PipelineSigningKMSKeyId}
+                  Resource:
+                    !If
+                      - CreatePipelineSigningKMSKey
+                      - !Sub arn:aws:kms:${AWS::Region}:${AWS::AccountId}:key/${PipelineSigningKMSKey}
+                      - !If
+                        - PipelineSigningKeyIsARN
+                        - !Ref PipelineSigningKMSKeyId
+                        - !Sub arn:aws:kms:${AWS::Region}:${AWS::AccountId}:key/${PipelineSigningKMSKeyId}
```

Comment:
This isn't neccessary as the API call in the Agent can take either KMS Key identifer or ARN. So there is no need to do anything special for the ARN https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/kms#SignInput 

And there is a precedent in the API for taking either in the same field, so you can remove the fancy ARN detection logic.

Reply:
Hey @wolfeidau, thanks for getting back to us.  You are quite right that the agent will have no trouble accepting either the Key ID or full ARN format.  We put this fancy conditional here because [to specify a KMS key in an IAM policy statement, you must use its key ARN](https://docs.aws.amazon.com/kms/latest/developerguide/cmks-in-iam-policies.html), that's my understanding of the documentation at least.

I'm definitely open to suggestions to simplify this stanza though.